build/media_source/com_media/js/edit-images.es6.js

```diff
@@ -2,304 +2,372 @@
+    });
+
+    this.addHistoryPoint = this.addHistoryPoint.bind(this);
+    this.createImageContainer = this.createImageContainer.bind(this);
+    this.activate = this.activate.bind(this);
+    this.Reset = this.Reset.bind(this);
+    this.Undo = this.Undo.bind(this);
+    this.Redo = this.Redo.bind(this);
+    this.createProgressBar = this.createProgressBar.bind(this);
+    this.updateProgressBar = this.updateProgressBar.bind(this);
+    this.removeProgressBar = this.removeProgressBar.bind(this);
+    this.upload = this.upload.bind(this);
+
+    // Create history entry
+    window.addEventListener('mediaManager.history.point', this.addHistoryPoint.bind(this));
   }
 
-  // Initiate the registry
-  Joomla.MediaManager.Edit.original = {
-    filename: options.uploadPath.split('/').pop(),
-    extension: options.uploadPath.split('.').pop(),
-    contents: `data:image/${options.uploadPath.split('.').pop()};base64,${options.contents}`,
-  };
-  Joomla.MediaManager.Edit.history = {};
-  Joomla.MediaManager.Edit.current = {};
+  /**
+   * Creates a history snapshot
+   * PRIVATE
+   */
+  addHistoryPoint() {
+    if (this.original !== this.current) {
+      const key = Object.keys(this.history).length;
+      if (this.history[key] && this.history[key - 1]
+        && this.history[key] === this.history[key - 1]) {
+        return;
+      }
+      this.history[key + 1] = this.current;
+    }
+  }
 
-  const activate = (name, data) => {
+  /**
+   * Creates the images for edit and preview
+   * PRIVATE
+   */
+  createImageContainer(data) {
     if (!data.contents) {
-      return;
+      throw new Error('Initialization error "edit-images.js"');
     }
-    // Create the images for edit and preview
-    const baseContainer = document.getElementById('media-manager-edit-container');
-    const editContainer = document.createElement('div');
-    const previewContainer = document.createElement('div');
-    const imageSrc = document.createElement('img');
-    const imagePreview = document.createElement('img');
-
-    baseContainer.innerHTML = '';
-    imageSrc.src = data.contents;
-    imageSrc.id = 'image-source';
-    imageSrc.style.maxWidth = '100%';
-    imagePreview.src = data.contents;
-    imagePreview.id = 'image-preview';
-    imagePreview.style.maxWidth = '100%';
-    editContainer.classList.add('hidden');
-
-    editContainer.appendChild(imageSrc);
-    baseContainer.appendChild(editContainer);
-
-    previewContainer.appendChild(imagePreview);
-    baseContainer.appendChild(previewContainer);
+
+    this.imagePreview = document.createElement('img');
+    this.imagePreview.src = data.contents;
+    this.imagePreview.id = 'image-preview';
+    this.imagePreview.style.width = '100%';
+    this.imagePreview.style.height = 'auto';
+    this.imagePreview.style.maxWidth = '100%';
+    this.baseContainer.appendChild(this.imagePreview);
+  }
+
+  async activate(name) {
+    this.imagePreview = null;
+    this.createImageContainer(this.current.contents);
```

Comment:
```suggestion
    this.createImageContainer(this.current);
```

So this fixes the initialisation error. But then i get 4 copies of the image loaded at full screen. So it's not really a good fix :)